packages/shared/lib/db/migrations/20240322141605_undelete_records_reset_createdat.cjs

```diff
@@ -0,0 +1,37 @@
+const RECORDS_TABLE = '_nango_sync_data_records';
+exports.up = async function (knex) {
+    await knex.transaction((trx) => {
+        return trx
+            .raw(
+                `
+            CREATE OR REPLACE FUNCTION ${RECORDS_TABLE}_reset_created_at()
+            RETURNS TRIGGER AS $$
+            BEGIN
+                IF OLD.external_deleted_at IS NOT NULL AND NEW.external_deleted_at IS NULL THEN
+                    NEW.created_at = NOW();
+                    NEW.updated_at = NOW();
+                    NEW.external_is_deleted = FALSE;
+                END IF;
+                RETURN NEW;
+            END;
+            $$ LANGUAGE plpgsql;
```

Comment:
There is no way to have this logic in the code? I'm really not fond of sql logic, it's hard to debug and easy to forget